### ISSUE TITLE
False positive [305] Use shell only when shell functionality is required

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-lint#492]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible-lint 4.1.0
```

# Desired Behaviour

Passing

# Actual Behaviour (Bug report only)

```
- hosts: localhost
  tasks:
    - name: foo
      shell: env
      register: result
      changed_when: false
      args:
        warn: no
```
This command is valid and should pass all linting because "env" must run as shell to capture entire shell environment and

At this moment this is throwing two errors:
```
[305] Use shell only when shell functionality is required
```

**Comment by netzvieh** [comment]
Hi @ssbarnea, for me the output of `env` using the `command` module and `shell` module is equal. Are you sure you need `shell` to do what you are doing?

And there is the [env Lookup](https://docs.ansible.com/ansible/latest/plugins/lookup/env.html) which might also be useful.

**Comment by ssbarnea** [comment]
@netzvieh The output of env can be the same, but there is no guarantee. This is because some of the variables could be defined by the shell config/rc files. People may want to use one or another. 

Re `lookup`s, there do work only on controller so they have a very different use-case.

**Comment by HugoPoi** [comment]
I think its an "edge" case scenario, and the `# noqa [rule_id]` is here for this type of use-case ?

**Comment by ssbarnea** [comment]
Right `# noqa 305` is the right way to address this.
